### PR TITLE
CI: use circleci

### PR DESCRIPTION
**Pull request by stsewd**
- Replace travis with circleci
- Update the drivers to its latest versions
- Update selenium pytest
- Test with more versions of sphinx and python
- Inject the extra js data before the head tag, as latest versions of sphinx don't inject jquery with the tag that the test was using.
- Fix tests for firefox (this test has always been failing for me locally before moving to circle)

The problem with the test was that it was assuming that the inner real width/height was the same the given by the test, but this doesn't count with extra space generated for window borders, navigation bar, etc (`set_viewport_size` uses this too).
This was failing only for firefox. Also, the current test was just assuming the margins were the same size (dividing by 2), instead I'm just checking that the margins are almost the same size (since we want to check that the modal is in the middle).

### CODE REVIEW

**Comment by stsewd** [comment]
We are almost there :D only one test is falling on firefox, will check tomorrow.

File: tox.ini [hunk 1]
```diff
@@ -1,19 +1,26 @@
...
   sphinx20: Sphinx<2.1
   sphinx21: Sphinx<2.2
+  sphinx22: Sphinx<2.3
+  sphinx23: Sphinx<2.4
+  sphinx24: Sphinx<2.5
+  sphinx30: Sphinx<3.1
+  sphinx31: Sphinx<3.2
+  sphinx32: Sphinx<3.3
+  sphinx33: Sphinx<3.4
+  sphinxlatest: Sphinx
```
Comment (ericholscher) [inline tox.ini:23]: I think we can probably scope this down a bit and only test against the latest release of each version? Similar to python: `sphinx1, sphinx2, sphinx3`?

**Review by ericholscher — APPROVED** [review]
Good to have these running properly :)

File: tox.ini (hunk 1 above)
Comment (stsewd) [reply tox.ini:23]: I noticed that there was a "half-breaking" change in 2.4 and 2.5 (the jquery change). But I'm +1 on supporting the latest versions of each mayor release. Less time waiting.

**Pull request merged**